### PR TITLE
Fix double hostname findings with trailing dots in Fierce boefje

### PR DESCRIPTION
**Pull request by tobiasBDO**
## Changes
Remove trailing dot from Fierce boefje output in order to fix duplicate findings with trailing dots.

Stripping trailing dot in order to fix duplicates such as: 
`- subdomain.domain.nl `
`- subdomain.domain.nl.`

## Issue ticket number and link
None

### before:
image

### after:
image

## Checklist for author(s):
- [x] All the commits in this PR are properly PGP-signed and verified;
- [ ] This PR comes from a `feature` or `hotfix` branch, in line with our git branching strategy;
- [x] This PR is "bite-sized" and only focuses on a single issue, problem, or feature;
- [x] I am not reinventing the wheel: there is no high-quality library that already has this feature;
- [ ] I have changed the example `.env` files if I added, removed, or changed any config options, and I have informed others that they need to modify their `.env` files if required;
- [x] I have performed a self-review of my own code;
- [x] I have commented my code, particularly in hard-to-understand areas;
- [ ] I have made corresponding changes to the documentation, if necessary;
- [ ] I have written unit, integration, and end-to-end tests for the change that I made;

If a non-trivial PR:
- [ ] This PR is part of a milestone and has appropriate labels;
- [ ] This PR is properly linked to the project board (either directly or via an issue);
- [ ] I have added screenshots or some other proof that my code does what it is supposed to do;

```
## Checklist for functional reviewer(s):
- [ ] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [ ] I have checked out this branch, and successfully ran `make kat`;
- [ ] I have ran `make test-rf` and all end-to-end Robot Framework tests pass;
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended;
- [ ] I confirmed that there are no unintended functional regressions in this branch;

### What works:
* _bullet point + screenshot (if useful) per tested functionality_

### What doesn't work:
* _bullet point + screenshot (if useful) per tested functionality_

### Bug or feature?:
* _bullet point + screenshot (if useful) if it is unclear whether something is a bug or an intended feature._
```

```
## Checklist for code reviewer(s):
- [ ] The code passes the CI tests and linters;
- [ ] The code does not bypass authentication or security mechanisms;
- [ ] The code does not introduce any dependency on a library that has not been properly vetted;
- [ ] The code does not violate Model-View-Template and our other architectural principles;
- [ ] The code contains docstrings, comments, and documentation where needed;
- [ ] The code prioritizes readability over performance where appropriate;
- [ ] The code conforms to our agreed coding standards.
```

### CODE REVIEW

File: boefjes/boefjes/plugins/kat_fierce/normalize.py [hunk 1]
```diff
@@ -17,7 +17,11 @@ def run(normalizer_meta: NormalizerMeta, raw: Union[bytes, str]) -> Iterator[OOI
     yield internet
 
     for _, subdomain in results["subdomains"].items():
-        host = Hostname(name=subdomain["url"], network=internet.reference)
+        if subdomain["url"][-1:] == ".":
```
Comment (ammar92) [inline boefjes/boefjes/plugins/kat_fierce/normalize.py:20]:
```suggestion
        if subdomain["url"].endswith("."):
```

Comment (ammar92) [reply boefjes/boefjes/plugins/kat_fierce/normalize.py:20]: Another (cleaner approach) would be:
```python3
...
subdomain = subdomain["url"].rstrip(".")
host = Hostname(name=subdomain, network=internet.reference)
...
```

**Review by ammar92 — COMMENTED** [review]
Nice fix! Just a remark to improve this even more

File: boefjes/boefjes/plugins/kat_fierce/normalize.py (hunk 1 above)
Comment (tobiasBDO) [reply boefjes/boefjes/plugins/kat_fierce/normalize.py]: Thanks for the feedback @ammar92 I implemented these changes, the suggested as well as the cleaner approach but kept the if-else statement there.

**Review by ammar92 — APPROVED** [review]
No remarks!

**Comment by tobiasBDO** [comment]
> No remarks!

I do have a remark @ammar92 I just tested these changes and they break this boefje with this:
```
  File "/app/boefjes/boefjes/plugins/kat_fierce/normalize.py", line 24, in run
    sub_ip = subdomain["ip"]
TypeError: string indices must be integers
```

Because of `subdomain = subdomain["url"].rstrip(".")` and the later assignment of `sub_ip = subdomain["ip"]` on line 24.

I propose a change like this to ensure subdomain and its list items can still be called in the rest of the boefje:
```
host = Hostname(name=subdomain["url"].rstrip("."), network=internet.reference)
yield host

sub_ip = subdomain["ip"]
```

**Comment by ammar92** [comment]
> > No remarks!
> 
> I do have a remark @ammar92 I just tested these changes and they break this boefje with this:
> 
> ```
>   File "/app/boefjes/boefjes/plugins/kat_fierce/normalize.py", line 24, in run
>     sub_ip = subdomain["ip"]
> TypeError: string indices must be integers
> ```
> 
> Because of `subdomain = subdomain["url"].rstrip(".")` and the later assignment of `sub_ip = subdomain["ip"]` on line 24.
> 
> I propose a change like this to ensure subdomain and its list items can still be called in the rest of the boefje:
> 
> ```
> host = Hostname(name=subdomain["url"].rstrip("."), network=internet.reference)
> yield host
> 
> sub_ip = subdomain["ip"]
> ```

Ah yes good point! I wasn't aware that it was used as a dictionary object elsewhere 👍

**Review by dekkers — APPROVED** [review]

**Pull request merged**